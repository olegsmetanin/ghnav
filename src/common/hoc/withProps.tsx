export const withProps = <P extends object>(propsfab: () => P) => (Component: React.ComponentType<P>) =>
  class WithProps extends React.Component<P> {
    public render() {
      const props = { ...((this.props as any) as object), ...propsfab() }
      return <Component {...props} />
    }
  }
