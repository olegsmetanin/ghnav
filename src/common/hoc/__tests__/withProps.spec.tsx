import React from 'react'
import renderer from 'react-test-renderer'
import { withProps } from '../withProps'

describe('withProps', () => {
  it('withProps', () => {
    class Comp extends React.Component<any, {}> {
      render() {
        return <div>{JSON.stringify(this.props)}</div>
      }
    }

    const CompWithProps = withProps(() => ({
      newprop: 1
    }))(Comp)

    const component = renderer.create(<CompWithProps />)
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
