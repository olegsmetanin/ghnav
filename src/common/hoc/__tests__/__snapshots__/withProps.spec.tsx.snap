// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`withProps withProps 1`] = `
<div>
  {"newprop":1}
</div>
`;
